### DISCUSSION TITLE
Allow user to configure `glob.globmatch()` to disable forced `wcmatch.PATHNAME` globbing

### DISCUSSION THREAD

**Discussion by aldenc98** [facelessuser/wcmatch#202] [Q&A] [answered]
Hi,

First, this is a great module and provides some great functionality for me!

I have a request/question related to `glob.globmatch()`.  I am using this module in Python3.8 on Windows to perform globbing on arbitrary strings with `glob.GLOBSTAR` enabled.  However, I'm running into an issue with my flags getting overridden and changing the behavior.

I'm trying to do this:
`glob.globmatch('//test/my/path/MyFile.txt', '//test/**/MyFile.txt', flags=glob.GLOBSTAR)`
and I'm not getting any matches. 

I've tracked it down to the `flags` being modified to explicitly add the `wcmatch.PATHNAME` flag here:
https://github.com/facelessuser/wcmatch/blob/main/wcmatch/glob.py#L109-L110

However, this causes a problem later, because this causes `WcParse.win_drive_detect == True`, and causes `_get_win_drive()` to incorrect detect and remove `//test` as a Windows drive, and causes the `**` pattern to not get expanded correctly.

Is there a way that I can disable the force set of `wcmatch.PATHNAME` so I can use globstar patterns in this way?  Or this is a feature request?

Thank you!

**Comment by facelessuser** [comment, marked as answer]
So, let me ask, are you wanting to match Unix style paths on Windows? If so, maybe this helps: https://facelessuser.github.io/wcmatch/glob/#forceunix.

You do not want to disable disable pathnames, that is what allows you to do paths vs just an fnmatch filename.

Wcmatch adapts itself to the system you are working on, and if Windows, it will use Windows logic for paths. If you are using something that does not touch the filesystem though, like globmatch, you force a Unix or Windows logic and match arbitrary paths.

EDIT: fixed typos

**Comment by aldenc98** [comment]
Oh, this totally worked for me!  The paths I'm matching are Perforce paths, which aren't strictly Unix-style (leading `//`, `...` to represent globstar-like matching, among other differences), but they're close enough to work with the Unix-style match.  Thank you for the quick turnaround!

**Comment by facelessuser** [comment]
No problem. Sorry for the somewhat broken reply. I was replying on my phone, so some of the grammar is wrong.

**Reply by facelessuser** [reply]
Moved to discussions.
